rust/datafusion/src/physical_plan/sort.rs

```diff
@@ -141,7 +141,10 @@ fn sort_batches(
     batches: &Vec<RecordBatch>,
     schema: &SchemaRef,
     expr: &[PhysicalSortExpr],
-) -> ArrowResult<RecordBatch> {
+) -> ArrowResult<Option<RecordBatch>> {
```

Comment:
Note that I did not use `create_empty_record_batch` to create a 0 row record batch (the approach taken in hash_aggregrate.rs)  https://github.com/apache/arrow/blob/master/rust/datafusion/src/physical_plan/hash_aggregate.rs#L827, as the approach in this PR will support any data type now or in the future.